tests/test_contrib_utils.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2017 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Test case for rebuilding access tokens."""
+
+import pytest
+from invenio_db import db
+
+from invenio_oauthclient.contrib.utils import rebuild_access_tokens
+from invenio_oauthclient.models import RemoteToken
+
+
+def test_rebuilding_access_tokens(models_fixture):
+    """Test rebuilding access tokens with random new SECRET_KEY."""
+    app = models_fixture
+    old_secret_key = app.secret_key 
+
+    datastore = app.extensions['invenio-accounts'].datastore
+    existing_email = 'existing@inveniosoftware.org'
+    user = datastore.find_user(email=existing_email)
+
+    # Creating a new remote token and commiting to the db
+    test_token = 'mytoken'
+    with db.session.begin_nested():
+        rt = RemoteToken.create(user.id, 'testkey', test_token, app.secret_key)
+        db.session.add(rt)
+    db.session.commit()
+
+    # Changing application SECRET_KEY
+    app.secret_key = "NEW_SECRET_KEY"
+
+    # Asserting the decoding error occurs with the stale SECRET_KEY
+    with pytest.raises(UnicodeDecodeError):
+        RemoteToken.query.all()
+
+    rebuild_access_tokens(app, old_secret_key)
+    remote_token = RemoteToken.query.all()[0]
```

Comment:
The returned first RemoteToken is not necessarily the one you expect. It would be better to query for it specifically.